cli/src/pcluster/templates/cdk_builder.py

```diff
@@ -12,15 +12,151 @@
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # TODO add S3 bucket
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=imagebuild.config.build_config.terminate_on_failure,
+            instance_types=["p2.xlarge", "c5.xlarge"] if archecture == "x86_64" else ["m6g.xlarge"],
+            key_pair=imagebuild.config.build_config.key_pair_name,
+        )
+
+        # Component
+        if imagebuild.config.dev_settings_config and imagebuild.config.dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/official_ami_build_component.yml",
+            )
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/custom_ami_build_component.yml",
+            )
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1",
+            version="0.0.1",
+            parent_image=core.Fn.sub(
+                "arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/amazon-linux-2-x86/x.x.x"
+                if not imagebuild.config.build_config.parent_image
+                else imagebuild.config.build_config.parent_image
+            ),
+            components=[
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ],
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
```

Comment:
let's define the ebs outside the block_device_mappings list

Reply:
Done